src/main/kotlin/no/skatteetaten/aurora/boober/service/OpenShiftObjectGenerator.kt

```diff
@@ -89,15 +98,31 @@ class OpenShiftObjectGenerator(
         return listOf(admin).addIfNotNull(view)
     }
 
+
+    //TODO: This is only used in tests. Should we have it here?
     fun generateDeploymentConfig(deployId: String, deploymentSpec: AuroraDeploymentSpec, provisioningResult: ProvisioningResult? = null): JsonNode? =
             withLabelsAndMounts(deployId, deploymentSpec, provisioningResult) { labels, mounts ->
                 generateDeploymentConfig(deploymentSpec, labels, mounts)
             }
 
     fun generateDeploymentConfig(auroraDeploymentSpec: AuroraDeploymentSpec,
                                  labels: Map<String, String>,
-                                 mounts: List<Mount>?): JsonNode? =
-            DeploymentConfigGenerator(mapper, velocityTemplateJsonService).create(auroraDeploymentSpec, labels, mounts)
+                                 mounts: List<Mount>?): JsonNode? {
+
+        if (auroraDeploymentSpec.deploy == null) {
+            return null
+        }
+        val deployment = auroraDeploymentSpec.applicationPlatform.handler.handleAuroraDeployment(auroraDeploymentSpec, labels, mounts)
+
+
```

Comment:
Can these generators be Objects now? Or how should we model them? They can also just be functions i guess?